libs/agno/agno/models/base.py

```diff
@@ -1542,11 +1618,55 @@ def format_function_call_results(
         self, messages: List[Message], function_call_results: List[Message], **kwargs
     ) -> None:
         """
-        Format function call results.
+        Format function call results and add follow-up user messages for generated media if needed.
         """
         if len(function_call_results) > 0:
             messages.extend(function_call_results)
 
+            if self.supports_media_in_tool_messages:
```

Comment:
I don't get this. Don't we have to split it regardless and it is up to the model itself to ignore it? We have that in lots of models, if they don't support images, they just log a warning and ignore that message.